mlos_bench/mlos_bench/config/schemas/services/remote/azure/azure-saas-service-subschema.json

```diff
@@ -0,0 +1,73 @@
+{
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "$id": "https://raw.githubusercontent.com/microsoft/MLOS/main/mlos_bench/mlos_bench/config/schemas/services/remote/azure/azure-saas-service-subschema.json",
+    "title": "mlos_bench Azure Flex DB configuration Service config",
+    "description": "config for an mlos_bench Azure Flex DB Service configuration",
+    "type": "object",
+    "properties": {
+        "class": {
+            "enum": [
+                "mlos_bench.services.remote.azure.AzureSaaSConfigService",
+                "mlos_bench.services.remote.azure.azure_saas.AzureSaaSConfigService"
+            ]
+        },
+        "config": {
+            "type": "object",
+            "allOf": [
+                {
+                    "$ref": "../../common-defs-subschemas.json#/$defs/poll_configs"
+                },
+                {
+                    "type": "object",
+                    "properties": {
+                        "subscription": {
+                            "description": "Azure subscription id (typically provided in the global config in order to omit from source control).",
+                            "type": "string"
+                        },
+                        "resourceGroup": {
+                            "description": "The name of the resource group to place the deployment in (typically provided in the global config in order to omit from source control).",
+                            "type": "string"
+                        },
+                        "vmName": {
```

Comment:
Is `vmName` the right term to use for a FlexServer instance?  Maybe just `saas_instance_name` for flexibility?

(BTW, I've found that prefixing names so they're globally unique allows for more flexible overrides across services from the `globals` configs)

Reply:
Also, is this required?

Reply:
yeah, I used `vmName` to make this environment a drop-in replacement for our stack of local+remote+local environments for MySQL.. We can rename it later. And yes, `vmName` *is* required - good catch, thank you!

Reply:
oops, never mind - `vmName` is _**not**_ required - we can pass it in `const_args` to the environment instead. (and that's another reason to keep the name for now)

Reply:
Maybe it would be better to rename all of them to `instance_name` instead of `vmName` then.  Call it future work is fine, but conceptually I think it would be cleaner.

Gets tricky though if you want to be able to mix and match both of these.  For instance, client VM for running `benchbase` or some other workload and one or more SaaS instances that act as the server backend(s).